app/services/game-overlay/index.ts

```diff
@@ -0,0 +1,396 @@
+
+const { BrowserWindow } = electron.remote;
+
+interface ICoordinates {
+  x: number;
+  y: number;
+}
+
+interface IWindowProperties {
+  chat: { position: ICoordinates; id: number };
+  recentEvents: { position: ICoordinates; id: number };
+}
+
+export type GameOverlayState = {
+  isEnabled: boolean;
+  isShowing: boolean;
+  isPreviewEnabled: boolean;
+  previewMode: boolean;
+  opacity: number;
+  windowProperties: IWindowProperties;
+};
+
+@InitAfter('UserService')
+@InitAfter('WindowsService')
+export class GameOverlayService extends PersistentStatefulService<GameOverlayState> {
+  @Inject() userService: UserService;
+  @Inject() customizationService: CustomizationService;
+  @Inject() windowsService: WindowsService;
+
+  static defaultState: GameOverlayState = {
+    isEnabled: false,
+    isShowing: false,
+    isPreviewEnabled: true,
+    previewMode: false,
+    opacity: 100,
+    windowProperties: {
+      chat: { position: null, id: null },
+      recentEvents: { position: null, id: null },
+    },
+  };
+
+  windows: {
+    chat: Electron.BrowserWindow;
+    recentEvents: Electron.BrowserWindow;
+    // overlayControls: Electron.BrowserWindow;
+  } = {} as any;
+
+  previewWindows: {
+    chat: Electron.BrowserWindow;
+    recentEvents: Electron.BrowserWindow;
+    // overlayControls: Electron.BrowserWindow;
+  } = {} as any;
+
+  overlayWindow: Electron.BrowserWindow;
+  onWindowsReady: Subject<Electron.BrowserWindow> = new Subject<Electron.BrowserWindow>();
+  onWindowsReadySubscription: Subscription;
+  lifecycle: LoginLifecycle;
+
+  commonWindowOptions = {};
+
+  async initialize() {
+    if (!this.state.isEnabled) return;
+
+    this.lifecycle = await this.userService.withLifecycle({
+      init: this.initializeOverlay,
+      destroy: this.destroyOverlay,
+      context: this,
+    });
+  }
+
+  async initializeOverlay() {
+    overlay.start();
+
+    this.onWindowsReadySubscription = this.onWindowsReady
+      .pipe(
+        take(Object.keys(this.windows).length),
+        delay(5000), // so recent events has time to load
+      )
+      .subscribe({ complete: () => this.createWindowOverlays() });
+
+    this.assignCommonWindowOptions();
+    this.createBrowserWindows();
+    this.createPreviewWindows();
+    await this.configureWindows();
+  }
+
+  assignCommonWindowOptions() {
+    const [containerX, containerY] = this.getWindowContainerStartingPosition();
+    this.commonWindowOptions = {
+      backgroundColor: this.customizationService.themeBackground,
+      show: false,
+      frame: false,
+      width: 300,
+      height: 300,
+      x: containerX,
+      y: containerY,
+      skipTaskbar: true,
+      thickFrame: false,
+      resizable: false,
+      webPreferences: { nodeIntegration: false, contextIsolation: true, offscreen: true },
+    };
+  }
+
+  createBrowserWindows() {
+    this.overlayWindow = new BrowserWindow({
+      ...this.commonWindowOptions,
+      height: 600,
+      width: 600,
+    });
```

Comment:
Is this window used by anything?  This seems like dead code unless I'm missing something.

Reply:
it's the parent of the other windows